### PR TITLE
Add 2 more VCPKG patches for OSS-fuzz

### PR DESCRIPTION
**Pull request by luckychess**
### Description of the Change
Pull request adds 2 more patches for vcpkg to make possible building fuzzing targets inside OSS-Fuzz image. They provide a workaround for building some problematic dependencies (boost-locale and grpc).

### Benefits
Ready to open pull request to the OSS-Fuzz!

### Possible Drawbacks 
None?

### Usage Examples or Tests 
Ask @luckychess to explain how to build everything inside the OSS-Fuzz image.

### CODE REVIEW

File: vcpkg/VCPKG_BOOST_LOCALE_DEPS_LIST [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+boost-iterator:
+boost-modular-build-helper:
+boost-smart-ptr:
+boost-static-assert:
+boost-system:
+boost-thread:
+boost-type-traits:
+boost-unordered:
+boost-vcpkg-helpers:
+libiconv:
```
Comment (igor-egorov) [inline vcpkg/VCPKG_BOOST_LOCALE_DEPS_LIST]: Why libiconv is put into VCPKG_BOOST_LOCALE_DEPS_LIST ? How it is related to BOOST deps?

File: vcpkg/patches/oss/0003-vcpkg-dependencies-sanitizer.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+index 68d90470..7508cd46 100644
+--- a/scripts/toolchains/linux.cmake
++++ b/scripts/toolchains/linux.cmake
+@@ -17,8 +17,8 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
+     string(APPEND CMAKE_C_FLAGS_RELEASE_INIT " ${VCPKG_C_FLAGS_RELEASE} ")
+     string(APPEND CMAKE_CXX_FLAGS_RELEASE_INIT " ${VCPKG_CXX_FLAGS_RELEASE} ")
+ 
+-    string(APPEND CMAKE_C_FLAGS_INIT " ${CMAKE_C_FLAGS} ")
+-    string(APPEND CMAKE_CXX_FLAGS_INIT " ${CMAKE_CXX_FLAGS} ")
++    string(APPEND CMAKE_C_FLAGS_INIT " ${CMAKE_C_FLAGS} -fsanitize=$ENV{SANITIZER} ")
```
Comment (igor-egorov) [inline vcpkg/patches/oss/0003-vcpkg-dependencies-sanitizer.patch]: Why -fsanitize is not enclosed by "if"? 
What will happen when $ENV{SANITIZER} is not set?

File: vcpkg/patches/oss/0002-vcpkg-dependencies-flags.patch [hunk 3]
```diff
@@ -0,0 +1,14 @@
+diff --git a/scripts/toolchains/linux.cmake b/scripts/toolchains/linux.cmake
+index 70f224da..68d90470 100644
+--- a/scripts/toolchains/linux.cmake
++++ b/scripts/toolchains/linux.cmake
+@@ -17,6 +17,9 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
```
Comment (igor-egorov) [inline vcpkg/patches/oss/0002-vcpkg-dependencies-flags.patch]: What is the meaning of _CMAKE_IN_TRY_COMPILE ?

File: vcpkg/VCPKG_BOOST_LOCALE_DEPS_LIST (hunk 1 above)
Comment (luckychess) [reply vcpkg/VCPKG_BOOST_LOCALE_DEPS_LIST]: This file contains a list of `boost-locale` dependencies (you can check them at https://github.com/microsoft/vcpkg/blob/master/ports/boost-locale/CONTROL). The actual reason why this list even exists is that `boost-locale` is broken - can't be built under sanitizers.
So the whole idea is next: build all dependencies under sanitizers, then build `boost-locale` without them and then build the rest with sanitizers again.

File: vcpkg/patches/oss/0002-vcpkg-dependencies-flags.patch (hunk 3 above)
Comment (luckychess) [reply vcpkg/patches/oss/0002-vcpkg-dependencies-flags.patch]: It's hard to say. I have to modify build flags globally for all packages but vcpkg doesn't have a way to do it (https://github.com/Microsoft/vcpkg/blob/master/docs/about/faq.md#can-i-use-my-ownspecific-flags-for-rebuilding-libs). So I found a workaround: https://github.com/Microsoft/vcpkg/issues/3577.

File: vcpkg/patches/oss/0003-vcpkg-dependencies-sanitizer.patch (hunk 2 above)
Comment (luckychess) [reply vcpkg/patches/oss/0003-vcpkg-dependencies-sanitizer.patch]: To be honest, I don't really care. It's intended to work only in a very specific environment (OSS-Fuzz container) which guarantees for specific environment variables are set.

**Pull request merged**